demisto_sdk/commands/content_graph/objects/pack.py

```diff
@@ -113,11 +121,63 @@ class PackMetadata(BaseModel):
     preview_only: Optional[bool] = Field(None, alias='previewOnly')
 
 
-class Pack(BaseContent, PackMetadata):
+class Pack(BaseContent, PackMetadata, content_type=ContentType.PACK):
     path: Path
     contributors: Optional[List[str]] = None
     content_items: PackContentItems = Field(alias='contentItems', exclude=True)
     relationships: Relationships = Field(Relationships(), exclude=True)
 
+    def dump_metadata(self, path: Path) -> None:
+        metadata = self.dict(exclude={'path', 'node_id', 'content_type'})
+        metadata['contentItems'] = {}
+        for content_item in self.content_items:
+            if content_item.content_type == ContentType.TEST_PLAYBOOK:
+                continue
+            try:
+                metadata['contentItems'].setdefault(content_item.content_type.server_name, []).append(content_item.summary())
+            except NotImplementedError as e:
+                logger.debug(f'Could not add {content_item.name} to pack metadata: {e}')
+        with open(path, 'w') as f:
+            json.dump(metadata, f, indent=4)
+
+    def dump_readme(self, path: Path, marketplace: MarketplaceVersions) -> None:
+        shutil.copyfile(self.path / 'README.md', path)
+        if self.contributors:
+            fixed_contributor_names = [f' - {contrib_name}\n' for contrib_name in self.contributors]
+            contribution_data = CONTRIBUTORS_README_TEMPLATE.format(contributors_names=''.join(fixed_contributor_names))
+            with open(path, 'a+') as f:
+                f.write(contribution_data)
+        with open(path, 'r+') as f:
+            try:
+                text = f.read()
+                parsed_text = get_mp_tag_parser(marketplace).parse_text(text)
+                if len(text) != len(parsed_text):
+                    f.seek(0)
+                    f.write(parsed_text)
+                    f.truncate()
+            except Exception as e:
+                logger.error(f'Failed dumping readme: {e}')
```

Comment:
let's try to make this file with as less non-relevant logic in pack object as possible
for example we can move this to a classmethod of MarketplaceParser